### PR TITLE
Add Nexell/RK SoCs, fix RK3399 stepping info

### PR DESCRIPTION
**Pull request by ThomasKaiser**

### CODE REVIEW

File: data/socs.yml [hunk 1]
```diff
@@ -2153,3 +2153,138 @@
...
+  features:
+  - fp
+  - asimd
+  - aes
+  - pmull
+  - sha1
+  - sha2
+  - crc32
+  - cpuid
+  name: S5P6818
```
Comment (hrw) [inline data/socs.yml:2170]: I love cryptic names used by vendors. Samsung had chip with same name (or maybe it was the same one).

**Review by hrw — CHANGES_REQUESTED** [review]

**Review by hrw — CHANGES_REQUESTED** [review]
can you squash it with previous one?

File: data/socs.yml [hunk 2]
```diff
@@ -2153,3 +2153,138 @@
...
+  - evtstrm
+  - aes
+  - pmull
+  - sha1
+  - sha2
+  - crc32
+  - cpuid
+  name: ThunderX CN8890
+  vendor: Cavium
+- cores:
```
[the remark this answers is not in the document]
Comment (ThomasKaiser) [reply data/socs.yml:2280]: Oh, I thought the X-Gene 1 is `revision: 0x1, variant: 0x0` while the APM883208-X1 is `0x0, 0x0`.

Ah, now I understand. You provided contents of `/proc/cpuinfo` in https://github.com/hrw/arm-socs-table/commit/d528f259a7974526af9171f49b5fcf1b79f2d0e3 yourself so X-Gene 1 is not `revision: 0x1, variant: 0x0` but `revision: 0x0, variant: 0x0` instead.

Ok, then I remove the 'APM883208-X1' definition in `socs.yml` and adjust there the 'X-Gene 1' entry to `revision: 0x0`.  But then `cpu_cores.yml` also needs an adjustment, right?

    '0x0-0-0':
      aarch32: EL0-EL2
      name: X-Gene 1

[the remark this answers is not in the document]
Comment (hrw) [reply data/socs.yml:2280]: X-Gene is tricky as they (APM/Ampere) used same part id for 3 generations - only revision/variant differ.

Will try to dig X-Gene 1 machine out and check details again.

**Comment by ThomasKaiser** [comment]
Just to be sure. Are you waiting on some input from my side or is your 'try to dig X-Gene 1 machine out and check details again' the next step? :)

**Comment by hrw** [comment]
Merged things with exception of that ancient X-Gene entry.

I am sorry that it took insane amount of time.

**Pull request closed without merging**